server/src/main/java/org/jboss/as/server/deployment/module/MountHandle.java

```diff
@@ -53,6 +53,7 @@ public MountHandle(final Closeable handle) {
     public void close() {
         if (handle != null) {
             VFSUtils.safeClose(handle);
+            handle = null; // it is already closed. not need to close again during gc.
```

Comment:
I was trying to keep the finalize method because I gueess it is there for proper clean up when some subsystems are not closing the mount handle (not calling the close method). I'm not sure if there is any sceneario where this could happen (that is why I kept it). If this cannot happen for sure I will remove it.